src/aws_encryption_sdk/streaming_client.py

```diff
@@ -583,8 +600,8 @@ def _read_bytes(self, b):
         :param int b: Number of bytes to read
         :raises NotSupportedError: if content type is not supported
         """
-        _LOGGER.debug("%s bytes requested from stream with content type: %s", b, self.content_type)
-        if b <= len(self.output_buffer) or self.source_stream.closed:
+        _LOGGER.debug("%d bytes requested from stream with content type: %s", b, self.content_type)
+        if 0 <= b <= len(self.output_buffer) or self.source_stream.closed:
```

Comment:
if b were None here, would 0 <= b <= len(x) return false?

Reply:
In Python 3 it would raise a `TypeError`. In Python 2, `None` is always less than any number.

However, in `read()` we normalize all `b` values of `None` to `-1`, and we test that `None` is accepted here[1] (added in #98 where I incorrectly normalized to `None`). The user interface is the same between #98 and here, but the internal implementation is more correct for PEP 3116.

[1] https://github.com/aws/aws-encryption-sdk-python/blob/master/test/unit/test_streaming_client_encryption_stream.py#L241